### PR TITLE
Initial technical doc

### PR DESCRIPTION
**Pull request by jmoreira-valory**
## Proposed changes

Describe the big picture of your changes here to communicate to the maintainers why we should accept this pull request.

## Fixes

If it fixes a bug or resolves a feature request, be sure to link to that issue.

## Types of changes

What types of changes does your code introduce? (A **breaking change** is a fix or feature that would cause existing functionality and APIs to not work as expected.)
_Put an `x` in the box that applies_

- [ ] Non-breaking fix (non-breaking change which fixes an issue)
- [ ] Breaking fix (breaking change which fixes an issue)
- [ ] Non-breaking feature (non-breaking change which adds functionality)
- [ ] Breaking feature (breaking change which adds functionality)
- [ ] Refactor (non-breaking change which changes implementation)
- [ ] Messy (mixture of the above - requires explanation!)

## Checklist

_Put an `x` in the boxes that apply._

- [ ] I have read the [CONTRIBUTING](../blob/main/CONTRIBUTING.md) doc
- [ ] I am making a pull request against the `main` branch (left side). Also you should start your branch off our `main`.
- [ ] Lint and unit tests pass locally with my changes
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] I have locally run services that could be impacted and they do not present failures derived from my changes

## Further comments

If this is a relatively large or complex change, kick off the discussion by explaining why you chose the solution you did and what alternatives you considered, etc...

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -1,9 +1,9 @@
...
-An autonomous service implementing a decentralized Impact Evaluator built with the [Open Autonomy](https://docs.autonolas.network/open-autonomy/) framework. The service tracks mentions to [@autonolas](https://twitter.com/autonolas) on Twitter, assigns scores to them and writes those scores to a [Ceramic](https://ceramic.network/) stream. It then reads verified wallets (a mapping between Wwitter handles and Ethereum addresses) from another Ceramic stream, checks for users that have minted a token, and updates their token metadata according to their scores so NFT images are upgraded over time.
+An autonomous service implementing a decentralized Impact Evaluator built with the [Open Autonomy](https://docs.autonolas.network/open-autonomy/) framework. The service tracks mentions to [@autonolas](https://twitter.com/autonolas) on Twitter, assigns scores to them and writes those scores to a [Ceramic](https://ceramic.network/) stream. It then reads verified wallets (a mapping between Twitter handles and Ethereum addresses) from another Ceramic stream, checks for users that have minted a token, and updates their token metadata according to their scores so NFT images are upgraded over time.
```
Comment (oaksprout) [inline README.md:2]: Should be "mentions of" rather than "mentions to"

File: docs/index.md [hunk 2]
```diff
@@ -0,0 +1,162 @@
+![IEKit](images/iekit.svg){ align=left }
+The Impact Evaluator Kit (IEKit) is an enhanced version of the [CoordinationKit](https://docs.autonolas.network/product/coordinationkit/), which leverages [Ceramic streams](https://developers.ceramic.network/docs/advanced/standards/stream-programs/) to automate the tracking and rewarding of users' contributions on the ecosystem. The Autonolas Impact Evaluator (IE) is an agent service based on the IEKit. This service is designed to track contributions in the form of Twitter mentions of the Autonolas DAO ([@autonolas](https://twitter.com/autonolas)). The service implements three main features:
+
+1. **Monitor for new users' registrations.** Any user can register as a contributor to the DAO by minting an NFT (through the [associated contract](https://etherscan.io/address/0x02c26437b292d86c5f4f21bbcce0771948274f84)) and sending a Tweet linking their wallet address.
+
+2. **Monitor for users' contributions.** The service periodically scans for new mentions of @autonolas on Twitter. If the mentions come from registered contributing users, the service increments and update their score.
```
Comment (dvilelaf) [inline docs/index.md:6]: Actually, the scores are incremented for every user, but only those who minted and therefore have an associated ETH address will see their badges updated.

File: README.md (hunk 1 above)
Comment (oaksprout) [reply README.md:2]: Last sentence could be clearer as an ordered list.

```
It then:
1. reads verified wallets (a mapping between Twitter handles and Ethereum addresses) from another Ceramic stream,
2. checks for users that have minted a token, and
3. updates their token metadata according to their scores so NFT images are upgraded over time.
```

File: docs/index.md [hunk 3]
```diff
@@ -0,0 +1,162 @@
+![IEKit](images/iekit.svg){ align=left }
+The Impact Evaluator Kit (IEKit) is an enhanced version of the [CoordinationKit](https://docs.autonolas.network/product/coordinationkit/), which leverages [Ceramic streams](https://developers.ceramic.network/docs/advanced/standards/stream-programs/) to automate the tracking and rewarding of users' contributions on the ecosystem. The Autonolas Impact Evaluator (IE) is an agent service based on the IEKit. This service is designed to track contributions in the form of Twitter mentions of the Autonolas DAO ([@autonolas](https://twitter.com/autonolas)). The service implements three main features:
+
+1. **Monitor for new users' registrations.** Any user can register as a contributor to the DAO by minting an NFT (through the [associated contract](https://etherscan.io/address/0x02c26437b292d86c5f4f21bbcce0771948274f84)) and sending a Tweet linking their wallet address.
```
Comment (dvilelaf) [inline docs/index.md:4]: This has not been implemented yet as it was not part of the grant. For the demo I wrote some dummy data to a stream but we are in the process of implementing it to use on Contribute (once it is replaced by IEKit).

File: docs/index.md [hunk 4]
```diff
@@ -0,0 +1,162 @@
+![IEKit](images/iekit.svg){ align=left }
+The Impact Evaluator Kit (IEKit) is an enhanced version of the [CoordinationKit](https://docs.autonolas.network/product/coordinationkit/), which leverages [Ceramic streams](https://developers.ceramic.network/docs/advanced/standards/stream-programs/) to automate the tracking and rewarding of users' contributions on the ecosystem. The Autonolas Impact Evaluator (IE) is an agent service based on the IEKit. This service is designed to track contributions in the form of Twitter mentions of the Autonolas DAO ([@autonolas](https://twitter.com/autonolas)). The service implements three main features:
```
Comment (oaksprout) [inline docs/index.md:2]: Should this be called Autonolas Impact Evaluator? Seems pretty official for something that's currently a demo. Really it's just an autonomous service which is powering the IEKit demo app.

**Review by oaksprout — CHANGES_REQUESTED** [review]

File: docs/index.md [hunk 5]
```diff
@@ -0,0 +1,162 @@
...
+
+    3. Create an `.env` file with the required environment variables, modifying its values to your needs.
+
+        ```bash
+        ETHEREUM_LEDGER_RPC=https://goerli.infura.io/v3/<infura_api_key>
+        DYNAMIC_CONTRIBUTION_CONTRACT_ADDRESS=0x7C3B976434faE9986050B26089649D9f63314BD8
+        WALLETS_STREAM_ID=<wallets_stream>
+        SCORES_STREAM_ID=<scores_stream>
+        CERAMIC_DID_SEED=<ceramic_seed_did>
+        CERAMIC_DID_STR=<ceramic_did_string>
```
Comment (dvilelaf) [inline docs/index.md:85]: the streams need to be created

File: docs/index.md (hunk 3 above)
Comment (dvilelaf) [reply docs/index.md:4]: Reads registered users from a Ceramic stream. This stream contains a mapping that links ethereum addresses to their corresponding Twitter id.

File: docs/index.md (hunk 2 above)
Comment (jmoreira-valory) [reply docs/index.md:6]: fixed.

File: README.md (hunk 1 above)
Comment (jmoreira-valory) [reply README.md:2]: fixed

File: docs/index.md (hunk 4 above)
Comment (jmoreira-valory) [reply docs/index.md:2]: fixed

File: docs/index.md [hunk 6]
```diff
@@ -0,0 +1,162 @@
...
+          {
+              "address": "0x14dC79964da2C08b23698B3D3cc7Ca32193d9955",
+              "private_key": "0x4bbbf85ce3377467afe5d46f804f221813b2bb87f24d81f60f1fcdbf7cbf4356"
+          }
+        ]
+        ```
+
+5. Prepare the environment and build the service deployment.
+
+    1. Create an API key for Google Spreadsheets API (you can follow [this guide](https://www.sharperlight.com/uncategorized/2022/04/06/accessing-the-google-sheets-api-via-sharperlight-query-builder/)).
```
[the remark this answers is not in the document]
Comment (jmoreira-valory) [reply docs/index.md:73]: fixed.

File: docs/index.md (hunk 5 above)
Comment (jmoreira-valory) [reply docs/index.md:85]: fixed.

File: docs/index.md (hunk 3 above)
Comment (jmoreira-valory) [reply docs/index.md]: fixed

**Review by dvilelaf — APPROVED** [review]

**Review by oaksprout — APPROVED** [review]

File: docs/index.md [hunk 7]
```diff
@@ -0,0 +1,166 @@
...
+        ]
+        ```
+
+5. Prepare the environment and build the service deployment.
+
+    1. Create a [Twitter API Bearer Token](https://developer.twitter.com/en/portal/dashboard).
+
+    2. Create a Ceramic Decentralized Identity (DID) using [Glaze](https://github.com/ceramicstudio/js-glaze).
+
+    3. Using the DID created in the previous step, create two Ceramic streams. You can follow [this tutorial](https://developers.ceramic.network/reference/stream-programs/tile-document/).
```
Comment (dvilelaf) [inline docs/index.md:77]:
```suggestion
    3. Using the DID created in the previous step, create two Ceramic streams. You can follow [this tutorial](https://developers.ceramic.network/reference/stream-programs/tile-document/). One of the streams should be empty and the service will write the scores to it. The other one should contain a mapping of Ethereum addresses to Twitter ids so the service kwows which address should be assigned the points from each Twitter id.
```    
{
   "<eth_address_1>": "twitter_id_1",
   "<eth_address_2>": "<twitted_id_2>"
}
```
```

File: docs/index.md
[the remark this answers is not in the document]
Comment (dvilelaf) [reply docs/index.md:1]: Already did it. It's working (I left a small suggestion)

**Pull request merged**